packages/ui-core/src/components/color-palette/color-palette.component.tsx

```diff
@@ -33,22 +40,29 @@ const ColorPalette = ({
   colorSuggestions,
   initialPickerColor = '#00FF00',
   maxNumberOfColors = 14,
+  scrollableParentRef,
 }: Props) => {
   const [isDragged, setIsDragged] = useState(false);
   const [colors, setColors] = useState(palette);
   const [activeColor, setActiveColor] = useState(null);
   const [addColorPickerOpen, setAddColorPickerOpen] = useState(false);
   const colorsOrderRef = useRef(colors);
   const sortableContainerRef = useRef(null);
-  const containerRef = useRef(null);
+  const colorPickerDropdownRef = useRef(null);
+  const addColorButtonRef = useRef(null);
+
+  const [colorPickerPosition, setColorPickerPosition] = useState({
```

Comment:
how about refactoring `useState`? one `useReducer` instead of 5x `useState`?

Reply:
👍  +1

Reply:
The similar approach is used in `<IntervalSlider />` component